core/src/main/java/org/apache/calcite/plan/Strong.java

```diff
@@ -149,14 +149,45 @@ public static boolean allStrong(List<RexNode> operands) {
     return operands.stream().allMatch(Strong::isStrong);
   }
 
-  /** Returns whether an expression is definitely not true. */
+  /** Returns whether an expression is definitely not true by either it being analyzed to be not
+   * true or null. */
   public boolean isNotTrue(RexNode node) {
+    return isPredicateNotTrue(node) || isNull(node);
+  }
+
+  /** Returns whether an expression is definitely not true. */
+  public boolean isPredicateNotTrue(RexNode node) {
```

Comment:
Why do you add this public method? Can it be merged into `isNotTrue`?

Reply:
We need this method since the methods below (allNotTrue, anyNotTrue) cannot rely on the isNotTrue due to the || isNull call.
I will make this method private

Reply:
made private.